### PR TITLE
Routine update of dom-related lib.d.ts

### PR DESCRIPTION
**Pull request by zhengbli**
Note: this update has tons of changes due to taking a new version of XML spec that shipped with windows 10 TH2. 

Fixes:
#5708
#6501 
#7218 
#7561 
#7562 
#7563 
#7197 
#6534

### CODE REVIEW

File: src/lib/webworker.generated.d.ts [hunk 1]
```diff
@@ -226,9 +230,9 @@ declare var Event: {
 }
 
 interface EventTarget {
-    addEventListener(type: string, listener: EventListenerOrEventListenerObject, useCapture?: boolean): void;
+    addEventListener(type: string, listener?: EventListenerOrEventListenerObject, useCapture?: boolean): void;
```
Comment (RReverser) [inline src/lib/webworker.generated.d.ts]: TIL `listener` is optional. Although not sure what's the point in allowing this :/

Comment (zhengbli) [inline src/lib/webworker.generated.d.ts]: As the library file is generated from the spec file we got from Edge team, this is a reflection of https://github.com/Microsoft/TSJS-lib-generator/blob/master/inputfiles/browser.webidl.xml#L3558

**Comment by zhengbli** [comment]
@mhegazy any comments?

**Comment by mhegazy** [comment]
:+1:

**Comment by myitcv** [comment]
@zhengbli as a result of this merge and per https://github.com/Microsoft/TSJS-lib-generator/issues/87 `declare var HTMLPictureElement` now appears twice in `dom.generated.d.ts`

As I mentioned in https://github.com/Microsoft/TSJS-lib-generator/issues/87 I would try and fix this and submit a PR but I can `build.sh` on Ubuntu. 

Any thoughts?

**Comment by zhengbli** [comment]
@myitcv it looks like the result of a bad git merge. I'll send a PR soon for a fix. Thanks for pointing out!

**Comment by sebilasse** [comment]
just btw:
`searchParams` seems to be missing in type `URL` as well …
https://developer.mozilla.org/en-US/docs/Web/API/URL

**Comment by zhengbli** [comment]
@sebilasse 
Indeed. Can you open a seperate issue to track it? You are also welcome to submit a PR to fix it at https://github.com/Microsoft/TSJS-lib-generator
The instructions are in the readme file in that repo. Thank you!

**Comment by sebilasse** [comment]
@zhengbli Hm - wouldn't it be possible to edit the dom.generated.d.ts "directly" ?
Otherwise I assume it would need to different steps (am I right?) :

1) Add a completely new interface [`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) to `addedTypes.json`

2) Add the property `URL.SearchParams` to `overridingTypes.json`

**Comment by zhengbli** [comment]
@sebilasse we never edit the `*.generated.d.ts` files by hand, because we periodically take a new version of the Edge idle and adopt the changes, manual changes would be very hard to maintain. 

Yes you are right, except the `URL.SearchParams` would be added to the `addedTypes.json` as well as a  new property. Thanks!

**Pull request merged**